dockstore-client/src/main/java/io/dockstore/client/cli/SwaggerUtility.java

```diff
@@ -43,19 +44,24 @@ private SwaggerUtility() {
                 new String[] { "BEARER" }, type);
     }
 
-    public static void unzipFile(File zipFile, File unzipDirectory) throws IOException {
+    public static void unzipFile(File zipFile, File unzipDirectory, boolean deleteZip) throws IOException {
         ZipFile zipFileActual = new ZipFile(zipFile);
-        zipFileActual.stream().forEach(zipEntry -> {
-            String fileName = zipEntry.getName();
-            File newFile = new File(unzipDirectory, fileName);
-            try {
-                newFile.getParentFile().mkdirs();
-                FileUtils.copyInputStreamToFile(zipFileActual.getInputStream(zipEntry), newFile);
-            } catch (IOException e) {
-                throw new RuntimeException(e);
+        zipFileActual.stream().forEach((ZipEntry zipEntry) -> {
+            if (!zipEntry.isDirectory()) {
+                String fileName = zipEntry.getName();
+                File newFile = new File(unzipDirectory, fileName);
+                try {
+                    newFile.getParentFile().mkdirs();
+                    FileUtils.copyInputStreamToFile(zipFileActual.getInputStream(zipEntry), newFile);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
             }
         });
-        FileUtils.deleteQuietly(zipFile);
+
+        if (deleteZip) {
```

Comment:
Should this be in a finally? If an an exception is thrown on line 57, this code will not get executed